scene/gui/licenses_dialog.cpp

```diff
@@ -0,0 +1,147 @@
+		SceneTree::get_singleton()->set_licenses_dialog_visible(false);
+		Node::get_viewport()->set_input_as_handled();
+	}
+}
+
+LicensesDialog::LicensesDialog() {
+	// Set on the highest layer, so that nothing else can draw on top.
+	set_layer(128);
+
+	set_process_unhandled_key_input(true);
+
+	MarginContainer *margin_container = memnew(MarginContainer);
+	margin_container->set_anchors_preset(Control::PRESET_FULL_RECT);
+	const float default_base_scale = margin_container->get_theme_default_base_scale();
+	const float default_font_size = margin_container->get_theme_default_font_size();
+	margin_container->add_theme_constant_override("margin_top", Math::round(20 * default_base_scale));
+	margin_container->add_theme_constant_override("margin_right", Math::round(20 * default_base_scale));
+	margin_container->add_theme_constant_override("margin_bottom", Math::round(20 * default_base_scale));
+	margin_container->add_theme_constant_override("margin_left", Math::round(20 * default_base_scale));
+	add_child(margin_container);
+
+	PanelContainer *panel_container = memnew(PanelContainer);
+	margin_container->add_child(panel_container);
+
+	MarginContainer *inner_margin_container = memnew(MarginContainer);
+	inner_margin_container->add_theme_constant_override("margin_top", Math::round(10 * default_base_scale));
+	inner_margin_container->add_theme_constant_override("margin_right", Math::round(10 * default_base_scale));
+	inner_margin_container->add_theme_constant_override("margin_bottom", Math::round(10 * default_base_scale));
+	inner_margin_container->add_theme_constant_override("margin_left", Math::round(10 * default_base_scale));
+	panel_container->add_child(inner_margin_container);
+
+	VBoxContainer *vbox_container = memnew(VBoxContainer);
+	vbox_container->add_theme_constant_override("separation", Math::round(10 * default_base_scale));
+	inner_margin_container->add_child(vbox_container);
+
+	Label *title_label = memnew(Label);
+	title_label->set_text(RTR("Third-party notices"));
+	title_label->add_theme_font_size_override("font_size", Math::round(1.333 * default_font_size));
+	title_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_CENTER);
+	vbox_container->add_child(title_label);
+
+	// Based on `editor_about.cpp` with references to TreeItem removed,
+	// as we only have the "All Components" view here. A preamble is also added.
+	String long_text = RTR("This project is powered by Godot Engine, which relies on a number of third-party free and open source libraries, all compatible with the terms of its MIT license. The following is an exhaustive list of all such third-party components with their respective copyright statements and license terms.") + "\n\n";
+
+	for (int component_index = 0; component_index < COPYRIGHT_INFO_COUNT; component_index++) {
+		const ComponentCopyright &component = COPYRIGHT_INFO[component_index];
+		String component_name = String::utf8(component.name);
+		String text = component_name + "\n";
```

Comment:
What is `text` for?

Reply:
It's a leftover from `editor_about.cpp` – I removed it.